core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -137,6 +138,22 @@ trait EntityDecoderInstances {
     EntityDecoder.decodeBy(MediaRange.`*/*`)(collectBinary)
   }
 
+  implicit def sourceDecoder[A](implicit W: EntityDecoder[A]): EntityDecoder[Process[Task,A]] =
+    new EntityDecoder[Process[Task,A]] {
+      override def decode(msg: Message): DecodeResult[Process[Task,A]] = {
+        val result = msg.body.flatMap{ byteVector =>
+          val fakeMessage = Request(body = Process.emit(byteVector))
+          Process.await(W.decode(fakeMessage).run) {
```

Comment:
Is this ever going to be able to return anything other than an error or a single-output `Process`?  I don't see any way for it to detect a boundary and loop back to process more `A`, which makes a `Process` seem like overkill.  It's also not clear to me how we could detect the boundaries of an arbitrary `A`.